packages/legend-query-builder/src/components/__tests__/QueryBuilderPostFilterPanel.test.tsx

```diff
@@ -1249,309 +1416,285 @@ test(
+      querySetup: {
+        entities: TEST_DATA__QueryBuilder_Model_SimpleRelationalWithDates,
+        modelCoverageAnalysisResult:
+          TEST_DATA__ModelCoverageAnalysisResult_SimpleRelationalWithDates,
+        mapping: 'model::RelationalMapping',
+        runtime: 'model::Runtime',
+      },
+      testClass: 'model::Person',
+      property: 'First Name',
+      propertyType: PrimitiveType.STRING,
+      valueToSet: 'test',
+      allowEmptyValue: true,
+    },
+    {
+      testName: `Query builder uses null as default value for numerical post-filter and shows validation error`,
+      querySetup: {
+        entities: TEST_DATA__QueryBuilder_Model_SimpleRelationalWithDates,
+        modelCoverageAnalysisResult:
+          TEST_DATA__ModelCoverageAnalysisResult_SimpleRelationalWithDates,
+        mapping: 'model::RelationalMapping',
+        runtime: 'model::Runtime',
+      },
+      testClass: 'model::Person',
+      property: 'Age',
+      propertyType: PrimitiveType.INTEGER,
+      valueToSet: '123',
+    },
+    {
+      testName: `Query builder uses null as default value for numerical post-filter on aggregate projection column and shows validation error`,
+      querySetup: {
+        entities: TEST_DATA__QueryBuilder_Model_SimpleRelationalWithDates,
+        modelCoverageAnalysisResult:
+          TEST_DATA__ModelCoverageAnalysisResult_SimpleRelationalWithDates,
+        mapping: 'model::RelationalMapping',
+        runtime: 'model::Runtime',
+      },
+      testClass: 'model::Person',
+      property: 'Age',
+      propertyType: PrimitiveType.INTEGER,
+      valueToSet: '123',
+      aggregate: 'count',
+    },
+    {
+      testName: `Query builder uses null as default value for date post-filter and shows validation error`,
+      querySetup: {
+        entities: TEST_DATA__QueryBuilder_Model_SimpleRelationalWithDates,
+        modelCoverageAnalysisResult:
+          TEST_DATA__ModelCoverageAnalysisResult_SimpleRelationalWithDates,
+        mapping: 'model::RelationalMapping',
+        runtime: 'model::Runtime',
+      },
+      testClass: 'model::Person',
+      property: 'Dob Date',
+      propertyType: PrimitiveType.DATE,
+      valueToSet: CUSTOM_DATE_PICKER_OPTION.TODAY,
+    },
+    {
+      testName: `Query builder uses null as default value for enum post-filter and shows validation error`,
+      querySetup: {
+        entities: TEST_DATA_QueryBuilder_QueryExecution_Entities,
+        modelCoverageAnalysisResult:
+          TEST_DATA__ModelCoverageAnalysisResult_QueryExecution_Entities,
+        mapping: 'model::RelationalMapping',
+        runtime: 'model::Runtime',
+      },
+      testClass: 'model::Firm',
+      property: 'Inc Type',
+      propertyType: Enumeration.prototype,
+      valueToSet: 'Corp',
+    },
+  ];
+
+describe(integrationTest('Post-filter default values are properly set'), () => {
+  test.each(POST_FILTER_PANEL_DEFAULT_VALUE_TEST_CASES)(
```

Comment:
I moved all the "null as default value" tests into this more generic test, except for boolean and list since those are a bit different.